agents/rust/aries-vcx-agent/tests/test_connection.rs

```diff
@@ -0,0 +1,85 @@
+#[macro_use]
+extern crate log;
+extern crate serde_json;
+extern crate aries_vcx_agent;
+extern crate aries_vcx;
+
+
+#[cfg(test)]
+#[cfg(feature = "agency_pool_tests")]
+mod integration_tests {
+    use aries_vcx;
+    use aries_vcx::agency_client::MessageStatusCode;
+    use aries_vcx::handlers::connection::mediated_connection::ConnectionState;
+    use aries_vcx_agent::{AgencyInitConfig, Agent as AriesAgent, InitConfig, PoolInitConfig, WalletInitConfig};
+    use aries_vcx::indy::ledger::pool::test_utils::{create_tmp_genesis_txn_file};
+    use aries_vcx::protocols::connection::invitee::state_machine::InviteeState;
+    use aries_vcx::protocols::connection::inviter::state_machine::InviterState;
+    use aries_vcx::utils::devsetup::init_test_logging;
+
+    pub async fn setup_pool() -> String {
+        let genesis_path = create_tmp_genesis_txn_file();
+        return genesis_path
+    }
+
+    pub async fn initialize() -> AriesAgent {
```

Comment:
I would move setup to a separate module, see #675. Would it be perhaps possible rebase this PR on top of #675 and (eventually) merge them together?